apply new, more generic analysis format (#8)
CERT-Polska/karton-yaramatcher: README.md

    "stage": "recognized",
    "kind": "dump"
}, {
    "type": "analysis"
}
```

> While `sample` type is self explanatory, the `analysis` type might be confusing. The `analysis` task is an output from

Reviewer: ```suggestion
While `sample` type is self explanatory, the `analysis` type might be a bit more complicated. The `analysis` task is an output from
```

one of sandboxes: `drakvuf-sandbox`, `cuckoo`, or `joesandbox`. Analysis is a `sample` with additional memory dumps
attached.

The `analysis` type task is expected to be in format:
```
task = Task(
    headers={"type": "analysis"}
    payload={
        "sample": <sample>,
        "dumps.zip": Resource.from_directory("dumps.zip", dumps_path.as_posix()),
        "dumps_metadata": [
            {"filename": <dump1_filename>, "base_address": <dump1_base_address>},
            {"filename": <dump2_filename>, "base_address": <dump2_base_address>},
            {"filename": <dump3_filename>, "base_address": <dump3_base_address>},
            [...]
        ],
    }
)
```

> where `dumps_metadata` contains information about filename and base address for every memory dump in `dumps.zip`. The

Reviewer: ```suggestion
where `dumps_metadata` contains information about filename and base address of every memory dump in `dumps.zip`. The
```

following attributes are:

> - `filename` which is relative path to the dumps.zip contents;

Reviewer: ```suggestion
- `filename` - a relative path to dump in the dumps.zip archive;
```

> - `base_address` which hex-encoded base address for dump (leading `0x` is supported)

Reviewer: ```suggestion
- `base_address` - hex-encoded dump base address (leading `0x` is supported)
```

You can specify multiple entries for the same file if the same memory dump was found on different base addresses.

> For the analysis type, yaramatcher runs rules against all dumps passed in `dumps.zip` resource and append result tags

Reviewer: > all dumps passed in `dumps.zip`

I think this isn't 100% correct? It still iterates over the items in `dumps_metadata`
